src/manager.rs

```diff
@@ -55,19 +55,19 @@ impl Manager {
     }
     */
 
-    pub fn add_node(&self, home_id:u32, secure: bool) -> Result<(), ()> {
+    pub fn add_node(&self, home_id:u32, secure: bool) -> Result<()> {
         res_to_result(unsafe {
             extern_manager::manager_add_node(self.ptr, home_id, secure)
-        })
+        }).or(Err(Error::APIError("Could not add a node")))
```

Comment:
This is actually not quite right. The failure to add a node message will be in the form of a notification. The only reason that this function will fail is because an invalid home_id was passed in. And that will be similar for all of the Manager functions. The ones that return false is due to invalid parameters, not that the action actually failed.